### ISSUE TITLE
when byobu is enabled, !(sudo cmd) doesn't capture output

### ISSUE THREAD

**Issue by eode** [xonsh/xonsh#2878]
## xonfig
```
+------------------+---------------------+
| xonsh            | 0.8.0               |
| Git SHA          | aca294b8            |
| Commit Date      | Oct 7 09:53:16 2018 |
| Python           | 3.6.6               |
| PLY              | 3.9                 |
| have readline    | True                |
| prompt toolkit   | None                |
| shell type       | readline            |
| pygments         | None                |
| on posix         | True                |
| on linux         | True                |
| distro           | Ubuntu              |
| on darwin        | False               |
| on windows       | False               |
| on cygwin        | False               |
| on msys2         | False               |
| is superuser     | False               |
| default encoding | utf-8               |
| xonsh encoding   | utf-8               |
| encoding errors  | surrogateescape     |
+------------------+---------------------+
```
## Expected Behavior
`!(sudo cmd)` works when `byobu` has been enabled via the `byobu-enable` command.

## Current Behavior
When `byobu-enable` has been called, and a session is active, using `!(sudo cmd)` does not capture output.  However, when byobu is run normally, capture occurs without error.

## Steps to Reproduce
* install byobu if not installed
* at command prompt, enter `byobu-enable`
* after confirmation, re-log.  A byobu session should be active.
* Attempt a captured command by doing `x = !(sudo ls)`
  * output prints to screen
  * captured command does not retain output.

**Comment by laloch** [comment]
I don't use byobu, yet `CommandPipeline` cannot capture output of any unthreaded subprocess:
```
laloch@uriel ~ $ __xonsh__.commands_cache.threadable_predictors['echo'] = lambda args: True                           
laloch@uriel ~ $ !(echo captured)                                                                                     
CommandPipeline(stdin=<_io.BytesIO object at 0x7fd72d8823b8>, stdout=<_io.BytesIO object at 0x7fd72ef95bf8>, stderr=<_io.BytesIO object at 0x7fd72ef95048>, pid=8838, returncode=0, args=['echo', 'captured'], alias=None, stdin_redirect=['<stdin>', 'r'], stdout_redirect=[8, 'wb'], stderr_redirect=[10, 'w'], timestamps=[1539906602.7268567, 1539906602.7337089], executed_cmd=['/usr/bin/echo', 'captured'], input=, output=captured
, errors=None)

laloch@uriel ~ $ __xonsh__.commands_cache.threadable_predictors['echo'] = lambda args: False                          
laloch@uriel ~ $ !(echo captured)                                                                                     
captured
CommandPipeline(stdin=None, stdout=None, stderr=None, pid=8914, returncode=0, args=['echo', 'captured'], alias=None, stdin_redirect=['<stdin>', 'r'], stdout_redirect=['<stdout>', 'a'], stderr_redirect=['<stderr>', 'r'], timestamps=[1539906616.6295648, 1539906616.6381817], executed_cmd=['/usr/bin/echo', 'captured'], input=, output=, errors=None)
```

**Comment by anki-code** [comment]
Closed with reference to #4214
